environment/src/main/java/com/sequenceiq/environment/network/service/SubnetIdProvider.java

```diff
@@ -31,9 +31,21 @@ public String provide(NetworkDto network, Tunnel tunnel, CloudPlatform cloudPlat
             LOGGER.debug("Check failed, returning null");
             return null;
         }
-        SubnetSelectionResult subnetSelectionResult = cloudPlatformConnectors.get(new CloudPlatformVariant(cloudPlatform.name(), cloudPlatform.name()))
-                .networkConnector()
-                .selectSubnets(new ArrayList<>(network.getSubnetMetas().values()), SubnetSelectionParameters.builder().withTunnel(tunnel).build());
+
+        NetworkConnector networkConnector = cloudPlatformConnectors
+                .get(new CloudPlatformVariant(cloudPlatform.name(), cloudPlatform.name()))
+                .networkConnector();
+        SubnetSelectionParameters subnetSelectionParameters = SubnetSelectionParameters
+                .builder()
+                .withTunnel(tunnel)
+                .build();
+
+        SubnetSelectionResult subnetSelectionResult = networkConnector
+                .filterSubnets(network.getSubnetMetas().values(), subnetSelectionParameters);
+        LOGGER.info("Filtered subnets are: {}", subnetSelectionResult);
+        subnetSelectionResult = networkConnector
+                .selectSubnets(subnetSelectionResult, subnetSelectionParameters);
+        LOGGER.info("Subnets after the selection are: {}", subnetSelectionResult);
```

Comment:
I would log subnets here, should have been logged in the subnet filter-select machinery.